tests/Resources/ti.database.test.js

```diff
@@ -0,0 +1,565 @@
+	// only if the result contains no fields/columns instead of the result containing no rows
+	it.skip('execute() returns null instead of empty result set', function () { // eslint-disable-line mocha/no-skipped-tests
+		should(Ti.Database.install).not.be.undefined;
+		should(Ti.Database.install).be.a.Function;
+
+		// Call install on a database file that doesn't exist. We should just make a new db with name 'category'
+		var db = Ti.Database.install('made.up.sqlite', 'category');
+
+		// Confirm 'db' is an object
+		should(db).be.a.Object;
+
+		var rows = db.execute('pragma table_info(\'category\');');
+
+		should(rows).be.null;
+
+		// Remove the 'category' database file
+		db.remove();
+
+		// Close the database (unnecessary as remove() does this for us)
+		db.close();
+	});
+
+	// If the source db file was not found, then install() must throw an exception.
+	it('install missing source db', function () {
+		let wasExceptionThrown = false;
+		try {
+			Ti.Database.install('BadFilePath.db', 'IShouldNotExist.db');
+		} catch (err) {
+			wasExceptionThrown = true;
+		}
+		should(wasExceptionThrown).be.true;
+	});
+
+	// Integer boundary tests.
+	// Verify we can read/write largest/smallest 64-bit int values supported by JS number type.
+	it('db read/write integer boundaries', function () {
+		const MAX_SIGNED_INT32 = 2147483647;
+		const MIN_SIGNED_INT32 = -2147483648;
+		const MAX_SIGNED_INT16 = 32767;
+		const MIN_SIGNED_INT16 = -32768;
+		const rows = [
+			Number.MAX_SAFE_INTEGER,
+			MAX_SIGNED_INT32 + 1,
+			MAX_SIGNED_INT32,
+			MAX_SIGNED_INT16 + 1,
+			MAX_SIGNED_INT16,
+			1,
+			0,
+			-1,
+			MIN_SIGNED_INT16,
+			MIN_SIGNED_INT16 - 1,
+			MIN_SIGNED_INT32,
+			MIN_SIGNED_INT32 - 1,
+			Number.MIN_SAFE_INTEGER
+		];
+
+		const dbConnection = Ti.Database.open('int_test.db');
+		dbConnection.execute('CREATE TABLE IF NOT EXISTS intTable(id INTEGER PRIMARY KEY, intValue INTEGER);');
+		dbConnection.execute('DELETE FROM intTable;');
+		for (let index = 0; index < rows.length; index++) {
+			dbConnection.execute('INSERT INTO intTable (id, intValue) VALUES (?, ?);', index, rows[index]);
+		}
+		const resultSet = dbConnection.execute('SELECT id, intValue FROM intTable ORDER BY id');
+		should(resultSet.rowCount).eql(rows.length);
+		for (let index = 0; resultSet.isValidRow(); resultSet.next(), index++) {
+			should(resultSet.field(1)).eql(rows[index]);
+		}
+		dbConnection.close();
+	});
+
+	describe.windowsMissing('#executeAsync()', () => {
+		it('is a function', () => {
+			const db = Ti.Database.open('execute_async.db');
+			try {
+				should(db.executeAsync).be.a.Function;
+			} finally {
+				db.close();
+			}
+		});
+
+		it('executes asynchronously', function (finish) {
+			this.timeout(5000);
+			const db = Ti.Database.open('execute_async.db');
+			// Execute a query to create a test table
+			db.executeAsync('CREATE TABLE IF NOT EXISTS testTable (text TEXT, number INTEGER)', err => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.database.test.js;
  Line: 416;
-->

- :warning: tests/Resources/ti.database.test.js line 416 – 'err' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)